AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -4295,41 +4297,27 @@ function Update-AzStorageAccountAuthForAES256 {
             -ResourceGroupName $ResourceGroupName -StorageAccountName $StorageAccountName -ErrorAction Stop
         $domain = $activeDirectoryProperties.DomainName
 
-        if ($adObject.ObjectClass -ine "computer") {
-            $message = "Removing object '$($adObject.DistinguishedName)' of type '$adObject.ObjectClass' from domain '$domain'." `
-                + " AES256 is only supported for computer objects."
-            Write-Verbose -Message $message
-
-            Remove-ADObject -Identity $adObject.DistinguishedName -Server $domain -Confirm:$false -ErrorAction Stop
-
-            $organizationalUnitDistinguishedName = $adObject.DistinguishedName.Substring($adObject.DistinguishedName.IndexOf(',') + 1)
-            $samAccountName = $adObject.SamAccountName.TrimEnd("$")
+        Set-StorageAccountDomainProperties `
+            -ADObjectName $adObjectName `
+            -ResourceGroupName $ResourceGroupName `
+            -StorageAccountName $StorageAccountName `
+            -Domain $domain `
+            -Force
 
-            $message = "Join storage account '$StorageAccountName' to domain '$domain'" `
-                + " as a computer object under '$organizationalUnitDistinguishedName'"
-            Write-Verbose -Message $message
+        switch($adObject.ObjectClass) {
+            "user" {
+                Write-Verbose -Message "Set AD user object '$($adObject.DistinguishedName)' to use AES256 for Kerberos authentication"
+                Set-ADUser -Identity $adObject.DistinguishedName -Server $domain `

```

Comment:
Suppose customer "joined" their storage account using an older version of the AzFilesHybrid module which did not set UPN.  Then they ran this cmdlet to upgrade to AES256.  Do we need to check that UPN exists and if not set it?

Reply:
Added a check for AD object, if it does not have UPN, will add it when set ADUser